### ISSUE TITLE
Error: from torch._C import _set_worker_signal_handlers, _update_worker_pids, 

### ISSUE THREAD

**Issue by AtefehBaniasadi** [saeed-anwar/RIDNet#19]
Hi, 

First of all, thank you for your work, when I wanted to test your model i faced this error 

from torch._C import _set_worker_signal_handlers, _update_worker_pids, 
ImportError: cannot import name '_update_worker_pids'

I thought the error may be related to pythorch version so I changed it but the error didn't resolve, can you please guide me on how I can address the problem?

Thanks in advance.

**Comment by mazzzystar** [comment]
@AtefehBaniasadi 
See https://github.com/CSAILVision/semantic-segmentation-pytorch/issues/157#issuecomment-492865334

**Comment by SuJiaKuan** [comment]
In [README](https://github.com/saeed-anwar/RIDNet/blob/master/README.md), the author says the model is built in PyTorch 0.4.0, PyTorch 0.4.1. Install PyTorch version 0.4.1 works for me.

If you use Conda, you can install PyTorch 0.4.1 by following command:

```bash
conda install pytorch=0.4.1 torchvision==0.2.1 cuda90 -c pytorch
```

**Comment by abbaskali** [comment]
if i have pytorch=1.10.0 is it possible to run this code or shall I return to pytorch =0.4.1 ?

**Comment by SuJiaKuan** [comment]
@abbaskali Please return to PyTorch 0.4.1 because the APIs changes a lot from version 0.X to 1.X

**Comment by abbaskali** [comment]
@SuJiaKuan  thank you a lot sir for your advice, could you help me with another question, I need Pytorch 1.10.0 so do you think is it possible to create another environment on my machine and install PyTorch 0.4.1 to run this code better? 
thank you once again

**Comment by SuJiaKuan** [comment]
@abbaskali Yes, it is possible. [Conda](https://docs.conda.io/en/latest/) is a very nice tool for Python packages and environment management. Under Conda, you can create separate environments for different purposes, for example, one environment for Pytorch 1.10.0 and one for Pytorch 0.4.1.

The [tutorial](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html) of Conda.

Example commands to create an environment for PyTorch 0.4.1:
```bash
conda create --name ridnet python=3.7
conda activate ridnet
conda install pytorch=0.4.1 torchvision==0.2.1 cuda90 -c pytorch
```

**Comment by abbaskali** [comment]
@SuJiaKuan thank you a lot for your great explanation, I appreciate it, sir, my cuda is 11.4 is it okay for "conda install pytorch=0.4.1 torchvision==0.2.1 cuda90 -c pytorch"?

**Comment by SuJiaKuan** [comment]
@abbaskali Yes, it is. The command also installs Cuda 9.0 (`cuda90`) that is separate from your own cuda (Cuda 11.4), so just feel free and try it.

**Comment by abbaskali** [comment]
@SuJiaKuan yes sir, i will do it, thank you for your help, God bless  you

**Comment by abbaskali** [comment]
anybody can help me I got this problem:
    from torch.utils.data import _utils
ImportError: cannot import name '_utils' from 'torch.utils.data' (C:\ProgramData\Miniconda3\envs\ridnet\lib\site-packages\torch\utils\data\__init__.py)

**Comment by abbaskali** [comment]
dear dr.@SuJiaKuan, i am sorry to bother you, this code is available for Ubuntu can I implement it on windows? or should I install the Ubuntu terminal? thank you

**Comment by SuJiaKuan** [comment]
@abbaskali I'm, a Linux user, so I'm not sure the code works well or not on Windows. Most popular Python / Deep Learning tools (e.g., Conda) support both Linux and Windows, and I guess it is very possible to run the project on Windows. From my perspective, most (Deep Learning) tools are native built on Linux, so it is recommended to learn and develop projects on Linux. :D

**Comment by abbaskali** [comment]
Dr. @SuJiaKuan thankful for you helping

**Comment by abbaskali** [comment]
dear dr.  @SuJiaKuan 
i got this problem could you help me please 
ModuleNotFoundError: No module named 'model.edsr'

**Comment by SuJiaKuan** [comment]
@abbaskali Do you run the code under directory `TestCode/code`?

**Comment by abbaskali** [comment]
> @abbaskali Do you run the code under directory `TestCode/code`?

yes sir, run it on this directory

**Comment by SuJiaKuan** [comment]
@abbaskali I don't see any file like `edsr.py` in [TestCode/code/model](https://github.com/saeed-anwar/RIDNet/tree/master/TestCode/code/model). If you create `edsr.py` by yourself, please make sure the file is located at the right directory. If not, please just delete the importing line and try it again.

**Comment by ttrfly** [comment]
> 任何人都可以帮助我我遇到了这个问题： from torch.utils.data import _utils ImportError: cannot import name ' _utils' from 'torch.utils.data' (C:\ProgramData\Miniconda3\envs\ridnet\lib\site-packages \火炬\ utils的\ data_中_初始化__的.py）

How did you solve this problem?

**Comment by SuJiaKuan** [comment]
> In [README](https://github.com/saeed-anwar/RIDNet/blob/master/README.md), the author says the model is built in PyTorch 0.4.0, PyTorch 0.4.1. Install PyTorch version 0.4.1 works for me.
> 
> If you use Conda, you can install PyTorch 0.4.1 by following command:
> 
> ```shell
> conda install pytorch=0.4.1 torchvision==0.2.1 cuda90 -c pytorch
> ```
@ttrfly See this comment.

**Comment by ttrfly** [comment]
I have done so, but the problem still arises.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "saeed-anwar/RIDNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年1月5日(星期三) 晚上9:50
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [saeed-anwar/RIDNet] Error: from torch._C import _set_worker_signal_handlers, _update_worker_pids,  (#19)





  
In README, the author says the model is built in PyTorch 0.4.0, PyTorch 0.4.1. Install PyTorch version 0.4.1 works for me.
 
If you use Conda, you can install PyTorch 0.4.1 by following command:
 conda install pytorch=0.4.1 torchvision==0.2.1 cuda90 -c pytorch
  
@ttrfly See this comment.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by SuJiaKuan** [comment]
@ttrfly Please check the PyTorch version by running:
```
pip show torch
```
